### PR TITLE
DEV-26949 Search users by username and full name

### PR DESCRIPTION
**Pull request by mrnyax**
Update documentation on the possibility to search users by username and full name

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1380,6 +1381,8 @@ The example response includes the pagination URLs in the `links` object so you k
 
 + Parameters
 
+    + username (string, optional) - Optional query parameter for searching by username
+    + fullName (string, optional) - Optional query parameter for searching by fullName
```
Comment (tkrisztian95) [inline apiary.apib:1385]:
```suggestion
    + fullName (string, optional) - Optional query parameter for searching by full name
```

File: apiary.apib [hunk 2]
```diff
@@ -1315,7 +1315,7 @@ Security:
 
 ## User Resource [/api/v1/user]
 
-### Get All Users [GET /api/v1/user{?sort,page,per_page}]
+### Get All Users [GET /api/v1/user{?sort,page,per_page,fullName,username}]
```
Comment (tkrisztian95) [inline apiary.apib:1318]: I would suggest to have the `username` param appended first:
```suggestion
### Get All Users [GET /api/v1/user{?sort,page,per_page,username,fullName}]
```

File: apiary.apib [hunk 3]
```diff
@@ -1326,6 +1326,10 @@ To display the list in ascending or descending order respectively, you can add a
 
 **Note:**
```
Comment (tkrisztian95) [inline apiary.apib:1328]: I would create a `Search` or `Filtering` paragraph with a short description and add these notes. What do you think? 

PS: Currently, this is in the `Sorting` section's notes.

File: apiary.apib [hunk 4]
```diff
@@ -1324,6 +1324,11 @@ This request retrieves a list of all users in the order that they were added.
 If you use the `sort` query parameter, you can only sort the list of users by 1 field. 
 To display the list in ascending or descending order respectively, you can add a `+` or `-`.
 
+**Searching/Filtering**
+
+You can use the `username` and `fullName` to search or filter users. The `username` and `fullName` query parameters
+are limited to 255 characters, are optional and cannot be `null`.
+
```
Comment (rbarbey) [inline apiary.apib:1331]: I would add that if both parameters set then users are found that have the given `username` or the given `fullName` but not both (OR and not AND)

**Review by rbarbey — CHANGES_REQUESTED** [review]

**Review by rbarbey — APPROVED** [review]

**Pull request merged**